newdle/client/src/components/App.js

```diff
@@ -3,33 +3,42 @@ import {useSelector} from 'react-redux';
 import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
 import {HomePage} from './home';
 import {CreationPage, CreationSuccessPage} from './creation';
-import {isLoginWindowOpen} from '../selectors';
+import {isLoginWindowOpen, getError} from '../selectors';
 import {AnswerPage} from './answer';
 import {SummaryPage} from './summary';
 import TopHeader from './TopHeader';
 import LoginPrompt from './login/LoginPrompt';
 import LoggingIn from './login/LoggingIn';
 import MyNewdles from './MyNewdles';
+import ErrorMessage from './ErrorMessage';
+
 import './App.module.scss';
 
 export default function App() {
   const loggingIn = useSelector(isLoginWindowOpen);
+  const error = useSelector(getError);
 
   return (
     <Router>
       <main>
         <TopHeader />
-        <Switch>
-          <Route exact path="/" component={HomePage} />
-          <Route exact path="/new" component={CreationPage} />
-          <Route exact path="/new/success" component={CreationSuccessPage} />
-          <Route exact path="/mine" component={MyNewdles} />
-          <Route path="/newdle/:code/summary" component={SummaryPage} />
-          <Route exact path="/newdle/:code/:partcode?" component={AnswerPage} />
-          <Route render={() => 'This page does not exist'} />
-        </Switch>
-        <LoginPrompt />
-        {loggingIn && <LoggingIn />}
+        {!error ? (
```

Comment:
Any reason in particular to use the negative formulation in here? I would do `error ? ...` to reduce confusion.

Reply:
Hm, no strong opinion here. Honestly I don't find it that confusing but true that it is not necessary to negate it here ;)